harness/determined/core/_train.py

```diff
@@ -67,41 +67,58 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
-    def report_training_metrics(
+    def report_trial_metrics(
         self,
-        steps_completed: int,
+        metric_type: str,
+        total_batches: int,
         metrics: Dict[str, Any],
         batch_metrics: Optional[List[Dict[str, Any]]] = None,
     ) -> None:
         """
-        Report training metrics to the master.
+        Report trial metrics to the master.
 
         You can include a list of ``batch_metrics``.  Batch metrics are not be shown in the WebUI
         but may be accessed from the master using the CLI for post-processing.
         """
 
-        body = {
-            "trial_run_id": self._run_id,
-            "steps_completed": steps_completed,
-            "metrics": {
-                "avg_metrics": metrics,
-            },
-        }
-        if batch_metrics is not None:
-            body["metrics"]["batch_metrics"] = batch_metrics  # type: ignore
-        logger.info(
-            f"report_training_metrics(steps_completed={steps_completed}, metrics={metrics})"
-        )
-        self._session.post(
-            f"/api/v1/trials/{self._trial_id}/training_metrics",
-            data=det.util.json_encode(body),
+        # CHECK: do we want this for training as well? same for batch_metrics?
```

Comment:
is this check comment still relevant?

Reply:
we should chat about this.
prior to this pr we skipped these for training metrics do you know if that was by design or an oversight? 
```
        serializable_metrics = self._get_serializable_metrics(metrics)
        reportable_metrics = {k: metrics[k] for k in serializable_metrics}
```
do we see us not wanting to skip unserializable metrics before serializing to send metrics over the wire? 🤔

Reply:
So if I'm understanding correctly, you are saying in the old metrics system we would skip metrics that don't serialize correctly? I don't know about the full conversation that led to that decision, but I would think this would be more confusing for the user to send certain invalid metrics and have them be silently ignored than to throw some kind of error/warning to say "this metric is invalid, please fix it"

Reply:
it looks like this is the case in `main` for some validation metrics but not training metrics (and I don't why the split happens) 
https://github.com/determined-ai/determined/blob/2ac22f00861bfc7c5700c47af5bde0f3f99949e5/harness/determined/core/_train.py#L142-L150

I have some guesses but let's take this conversation to #mlsys